### PR TITLE
[Ready for review] Drag and drop revision rows #170

### LINKED ISSUE #170: Enable re-ordering selected revisions
Allow users to drag+drop to reorder selected revisions.

### PR DESCRIPTION
**Pull request by NikolaDojic**
closes #170.
ready for review.

### CODE REVIEW

**Comment by NikolaDojic** [comment]
I don't have the right permission to add labels to the PRs, but you can consider this ready for review.

**Review by kimberlythegeek — CHANGES_REQUESTED** [review]
Thanks! I would ask that you add some sort of UI "affordance" to indicate to users that these items are draggable.

This is a resource on drag-and-drop affordances: https://www.nngroup.com/articles/drag-drop/

at the very least I would expect an icon that suggests the items can be dragged. I appreciate your contribution :smile:

**Comment by beatrice-acasandrei** [comment]
@NikolaDojic Thanks for your contributions! Could you update the PR so that it's no longer out-of-date (rebase with upstream/staging)?

**Comment by NikolaDojic** [comment]
thanks for the reviews :) 
@beatrice-acasandrei rebased,
@kimberlythegeek that is a good point, thanks, I've added a draggable icon indicator at the beginning of the draggable rows.

**Comment by beatrice-acasandrei** [comment]
While dragging the cursor's icon changes which might be confusing since it indicates you can't drop the item.

**Comment by NikolaDojic** [comment]
> While dragging the cursor's icon changes which might be confusing since it indicates you can't drop the item.

@beatrice-acasandrei 
hmm, I'm not sure that I follow. 
What is happening currently is that we have one cursor icon that is visible on hover, which is an open-hand icon that indicates that the element is draggable.
After the drag start event,  the cursor changes to a closed hand, which is the default browser behavior and it indicates that the item is being dragged (this is due to the `draggable="true"` attribute ). 
The cursor changes to an open hand after the element is dropped inside the table (since it is hovering over the row).

Can you help me understand which part is problematic? 
thanks :)

**Comment by NikolaDojic** [comment]
@beatrice-acasandrei thanks for the update.
hmm, interesting. It could be os related since I'm working on Arch Linux

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -49,11 +49,32 @@ const setConfidenceClassName = (confidenceText: string | null) => {
...
+): T[] => {
+  const newArray = [...array];
+  if (
+    index1 !== index2 &&
+    [index1, index2].every((index) => index < newArray.length && index >= 0)
+  ) {
+    [newArray[index1], newArray[index2]] = [newArray[index2], newArray[index1]];
+    return newArray;
+  }
+  console.warn(
```
Comment (beatrice-acasandrei) [inline src/utils/helpers.ts:65]: I assume `console.warn` was used for testing purposes and can be removed.

**Comment by kimberlythegeek** [comment]
Hi--unsure if I'm missing something, but I've just run this locally and the drag-and-drop does not seem to be working

**Review by kimberlythegeek — CHANGES_REQUESTED** [review]
Hi--unsure if I'm missing something, but I've just run this locally and the drag-and-drop does not seem to be working

**Comment by NikolaDojic** [comment]
> Hi--unsure if I'm missing something, but I've just run this locally and the drag-and-drop does not seem to be working

that is weird :/ 
Can you tell me more about your env? 
and maybe git log output?

here is a screen recording from my local branch, which is, according to git same as my remote branch.
The last commit has hash `c4657e583c994714760ea7c1ad571cb137005c98`
https://user-images.githubusercontent.com/24179802/210759257-f247be9c-cf3e-4f97-a689-7f5e9630b30e.mp4

**Comment by kimberlythegeek** [comment]
Hi--unsure what was causing the error before, but it appears to be working now!

**Review by kimberlythegeek — APPROVED** [review]

**Comment by esanuandra** [comment]
Hey @NikolaDojic, can you update de pr again please so we can merge this?

**Comment by NikolaDojic** [comment]
@esanuandra  updated

**Pull request merged**